example/fir_filter.dart

```diff
@@ -0,0 +1,123 @@
+/// SPDX-License-Identifier: BSD-3-Clause
+/// fir_filter.dart
+/// A basic example of a FIR filter
+///
+/// 2022 February 02
+/// Author: wswongat
+///
+import 'package:rohd/rohd.dart';
+import 'dart:io';
+
+class FirFilter extends Module {
+  // For convenience, map interesting outputs to short variable names for consumers of this module
+  Logic get out => output('out');
+
+  // This Fir Filter supports any width and depth, defined at runtime
+  final int bitWidth;
+  final int depth;
+  FirFilter(Logic en, Logic resetB, Logic clk, Logic inputVal, List<int> coef,
+      {this.bitWidth = 16, name = 'FirFilter'})
+      : depth = coef.length,
+        super(name: name) {
+    // Register inputs and outputs of the module in the constructor
+    // Module logic must consume registered inputs and output to registered outputs
+    if (depth < 1) {
+      // Depth Check
+      throw Exception('Depth parameter should more than 1');
+    }
+    // Add input/output port
+    en = addInput('en', en);
+    inputVal = addInput('inputVal', inputVal, width: bitWidth);
+    resetB = addInput('reset', resetB);
```

Comment:
Minor, but the name of the reset pin should be updated to 'resetB' as well

Reply:
Thank you for reminding me.
All fixed.